QuadPaySDK/PaymentWidget/PaymentWidget.swift

```diff
@@ -13,24 +13,61 @@ public final class PaymentWidget: UIView {
     
     let stackView = UIStackView()
     
+    private var request: ApiRequest<WidgetDataResource>?
+    
     @objc public var merchantId: String = "" {
         didSet{
-            MerchantService.shared.fetchMerchants(merchantId: merchantId){ (result) in
-                switch result {
-                case .success(_):
-                    self.paymentWidgetHeaderText.actualPaymentWidgetLabelText = "Split your order in 4 easy payments with Welcome Pay (powered by Zip)."
-                    self.paymentWidgetHeaderText.style()
-                    self.timelapseGraphView.depth = 0
-                    self.timelapseGraphView.drawTimelapseGraph()
-                case .failure(_):
-                    self.paymentWidgetHeaderText.actualPaymentWidgetLabelText = "Split your order in 4 easy payments with Zip."
-                    self.timelapseGraphView.depth = 3
-                    DispatchQueue.main.async {
-                        self.paymentWidgetHeaderText.style()
-                        self.timelapseGraphView.drawTimelapseGraph()
+            if(amount != "0"){
```

Comment:
If amount is set after the `merchantId` widget data won't be fetched.

Maybe a function `setWidgetData` to encapsulate the fetching and setting of the `maxFee` should be defined which can be called from both places.